Sources/Components/ReviewButtonView/ReviewButtonView.swift

```diff
@@ -0,0 +1,100 @@
+//
+//  Copyright © 2019 FINN AS. All rights reserved.
+//
+
+import UIKit
+
+public protocol ReviewButtonViewDelegate: AnyObject {
+    func reviewButtonView(_ reviewButtonView: ReviewButtonView, giveReviewWasTapped startReview: Bool)
+}
+
+public final class ReviewButtonView: UIView {
+    // MARK: - Public
+
+    public weak var delegate: ReviewButtonViewDelegate?
+    public var text: String = "" {
+        didSet {
+            reviewButtonControl.text = text
+        }
+    }
+
+    // MARK: - Private
+
+    private let reviewButtonControlHeight: CGFloat = 48
+    private let viewHeight: CGFloat = 64
+
+    private lazy var hairlineSeperator: UIView = {
+        let view = UIView(withAutoLayout: true)
+        view.backgroundColor = .sardine
+        return view
+    }()
+
+    private lazy var reviewButtonControl: ReviewButtonControl = {
+        let control = ReviewButtonControl()
+        control.delegate = self
+        return control
+    }()
+
+    // MARK: - Init
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        setup()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        setup()
+    }
+
+    // MARK: - Public methods
+
+    public func addToView(_ view: UIView) {
```

Comment:
Would it be possible to simply use `didlMoveToSuperview` instead?

Reply:
That's a nice one, will fix it.